src/pybind/mgr/ssh/module.py

```diff
@@ -1392,4 +1388,120 @@ def _remove_rbd_mirror(daemons):
         return self._get_services('rbd-mirror').then(_remove_rbd_mirror)
 
     def update_rbd_mirror(self, spec):
+        spec = NodeAssignment(spec=spec, get_hosts_func=self._get_hosts, service_type='rbd-mirror').load()
         return self._update_service('rbd-mirror', self.add_rbd_mirror, spec)
+
+
+class BaseScheduler(object):
+    """
+    Base Scheduler Interface
+
+    * requires a placement_spec
+
+    `place(host_pool)` needs to return a List[HostSpec, ..]
+    """
+
+    def __init__(self, placement_spec):
+        # type: (orchestrator.PlacementSpec) -> None
+        self.placement_spec = placement_spec
+
+    def place(self):
+        raise NotImplementedError
+
+
+class SimpleScheduler(BaseScheduler):
+    """
+    The most simple way to pick/schedule a set of hosts.
+    1) Shuffle the provided host_pool
+    2) Select from list up to :count
+    """
+    def __init__(self, placement_spec):
+        super(SimpleScheduler, self).__init__(placement_spec)
+
+    def place(self, host_pool, count=None):
+        # type: (List, Optional(int)) -> List
+        if not host_pool:
+            raise Exception('List of host candidates is empty')
+        host_pool = [HostSpec(x, '', '') for x in host_pool]
+        # shuffle for pseudo random selection
+        random.shuffle(host_pool)
```

Comment:
Until we have some info about the hosts, there isn't anything sensible we can do here.